### PR TITLE
Pobrien/changes for radar server

### PR DESCRIPTION
**Pull request by bolddane**
WIP: changes required for adding _radar_message_ to _radar_.  Also, some minor changes that will be used in _radar_client_

/cc @zendesk/zendesk-radar
### Steps to merge
- [ ] :+1: of the team
### References
- Jira link: https://zendesk.atlassian.net/browse/RADAR-574
### Risks
- None

### CODE REVIEW

**Comment by vanchi-zendesk** [comment]
it's very hard to figure out what is going on here without the corresponding server PR.

File: lib/message_request.js [hunk 1]
```diff
@@ -77,6 +78,18 @@ Request.buildUnsubscribe = function (scope) {
   return new Request(message);
 };
 
+Request.parse = function (data) {
+  try {
+    var message = JSON.parse(data);
+    return message;
```
Comment (vanchi-zendesk) [inline lib/message_request.js:84]: I would expect Request.parse to return a request.

File: lib/message_request.js [hunk 2]
```diff
@@ -77,6 +78,18 @@ Request.buildUnsubscribe = function (scope) {
   return new Request(message);
 };
 
+Request.parseMessageFromData = function (data) {
```
Comment (junosuarez) [inline lib/message_request.js:81]: I'm -1 to including a class method wrapping JSON.parse. I would prefer something with a signature like  `Request.createFromString(String) => RequestMessage` that included JSON.parse and instantiated a Request object, which it returned.

File: lib/message_request.js [hunk 3]
```diff
@@ -128,6 +145,14 @@ Request.prototype.getType = function () {
   return this.type;
 };
 
+Request.prototype.isValid = function () {
+  return this.valid;
+};
+
+Request.prototype.isOp = function (operation) {
+  return (this.getAttr('op') === operation);
```
Comment (junosuarez) [inline lib/message_request.js:153]: outer parentheses not needed

File: lib/message_response.js [hunk 4]
```diff
@@ -39,10 +40,15 @@ Response.prototype._validate = function () {
   return true;
 };
 
-Response.prototype.isValid = function () {
+// TODO: should probably be renamed isValidForSync
+Response.prototype.isValidForSync = function () {
```
Comment (junosuarez) [inline lib/message_response.js:44]: the difference in meaning between isValid and isValidForSync should be documented (it's not clear to me what this method does)

**Comment by junosuarez** [comment]
Closing, changes have been otherwise incorporated

**Pull request closed without merging**